ext/session/tests/session_start_read_and_close.phpt

```diff
@@ -0,0 +1,49 @@
+--TEST--
+Test session_start() with flag read_and_close
+--EXTENSIONS--
+session
+--SKIPIF--
+<?php include('skipif.inc'); ?>
+--FILE--
+<?php
+
+ob_start();
+
+$valuesEnablingReadAndClose = [true, 1, "1", "777", 777, -1];
+
+$valuesDisablingReadAndClose = ["true", false, "false", 0, "0", "no", "00000"];
+
+foreach ($valuesEnablingReadAndClose as $value) {
+    session_start(["read_and_close" => $value]);
+    var_dump(session_status() === PHP_SESSION_NONE);
+}
+
+foreach ($valuesDisablingReadAndClose as $value) {
+    session_start(["read_and_close" => $value]);
+    var_dump(session_status() === PHP_SESSION_ACTIVE);
+    session_write_close();
+}
+
+try {
+    session_start(["read_and_close" => 1.1]);
```

Comment:
What happens with a float like `1.0` ?

Reply:
The behavior is the same. Float is not allowed. I changed to `1.0` to have a more explicit test.